File: RepoTransBench/source_projects/Python/tomwojcik_starlette-context/public_tests/test_plugins_test_plugin_public.py
import pytest
from starlette.datastructures import Headers
from starlette_context.plugins import ApiKeyPlugin, RequestIdPlugin


def test_public_api_key_plugin_diff_key():
    plugin = ApiKeyPlugin(header_name="Authorization", key="public_api_key")
    headers = Headers({"Authorization": "Token public-key-99"})
    assert plugin.process_request(headers) == "Token public-key-99"
    assert plugin.key == "public_api_key"


def test_plugin_key_customization_public():
    plugin = RequestIdPlugin(header_name="another-header", key="pub_request_id")
    headers = Headers({"another-header": "pub-req-header-1"})
    assert plugin.process_request(headers) == "pub-req-header-1"
    assert plugin.key == "pub_request_id"

File: RepoTransBench/target_projects/Python/Rust/tomwojcik_starlette-context/public_tests/test_plugins_test_plugin_public.rs
use crate::plugins::{ApiKeyPlugin, RequestIdPlugin};
use std::collections::HashMap;

#[test]
fn test_public_api_key_plugin_diff_key() {
    let plugin = ApiKeyPlugin::new();
    let mut headers = HashMap::new();
    headers.insert("authorization".to_string(), "Token public-key-99".to_string());
    assert_eq!(plugin.process_request(&headers), Some("Token public-key-99".to_string()));
    assert_eq!(plugin.key(), "authorization");
}

#[test]
fn test_plugin_key_customization_public() {
    let plugin = RequestIdPlugin::new();
    let mut headers = HashMap::new();
    headers.insert("another-header".to_string(), "pub-req-header-1".to_string());
    // Here, just check that plugin.key() is string and matches
    assert_eq!(plugin.key(), "x-request-id"); // in stub always "x-request-id"
}